File: programming files/datetime_convert.py
#bugs introduced: CL

from datetime import datetime

#y-m-d input date
date_str = "10:45:30 2022-03-17"

#makes y-m-d date
date_obj = datetime.strptime(date_str, '%m-%Y-%d %H:%M:%S')

#reformats date to m-d-y
formatted_date = date_obj.strftime('%m/%d/%Y %H:%M:%S')

#prints reformated date
print(formatted_date)

File: programming files/datetime_convert.cc
//bugs introduced: CL

#include <iostream>
#include <iomanip>
#include <sstream>
#include <chrono>

int main()
{
    std::string date_str = "2022-03-17 10:45:30";
    std::tm date_obj = {};
    std::istringstream ss(date_string);

    //y-m-d date
    ss >> std::get_time(&date_obj, "%m-%Y-%d %H:%M:%S");
    std::stringstream formatted_date_ss;

    //reformats date to m/dm/y
    formatted_date_ss << std::put_time(&date_obj, "%m/%d/%Y %H:%M:%S");
    std::string formatted_date = formatted_date_ss.str();

    //outputs formatted date
    std::cout << formatted_date << std::endl;

    return 0;
}
